### PR TITLE
Lua strings format()

### LINKED ISSUE #3109: [Term Entry] Lua Strings format()
### Reason/inspiration (optional)

Please add the `format()` function as a new term entry as part of the string functions terms. Provide the entry in a markdown file and place under the file path `content/lua/concepts/strings/terms/format/format.md`

The entry should have the following:

� An introductory paragraph about what the function is and does.
� A syntax section with a pseudo code block with an explanation of it.
� An example showing the function in use.

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Brag2gr8**
### Description

Added term entry for Lua String function format() .
File Location: content\lua\concepts\strings\terms\format\format.md. Closes #3109

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/strings/terms/format/format.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  - 'Game Development'
+Tags:
+  - 'Strings'
+  - 'Functions'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`.format()`** is used to create formatted strings by replacing placeholders in a format string with specified values.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/format/format.md:15]:
```suggestion
The **`.format()`** method is used to create formatted strings by replacing placeholders with specified values.
```

File: content/lua/concepts/strings/terms/format/format.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+- `format`: A format string that defines the structure of the output string. It contains placeholders and formatting options.
+- `value1`, `value2`, etc. (optional): Values to be inserted into the placeholders within the format string.
+
+Here are some common placeholders and formatting options:
+
+`%s`: Placeholder for a string.
+`%d` or `%i`: Placeholder for an integer.
+`%f`: Placeholder for a floating-point number (decimal).
+`%x` or `%X`: Placeholder for a hexadecimal number.
+`%.nf`: Specify the number of decimal places (e.g., %.2f for two decimal places).
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/format/format.md:32]:
```suggestion
`%.nf`: Designates the number of decimal places (e.g. `%.2f` for two decimal places).
```

File: content/lua/concepts/strings/terms/format/format.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+`%c`: Placeholder for a character.
+`%%`: Literal % character.
+
+Here are some examples of how to use `string.format()` :
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/format/format.md:36]:
```suggestion
The following example demonstrates an implementation of `string.format()` :
```

Comment (caupolicandiaz) [reply content/lua/concepts/strings/terms/format/format.md:36]: move this line after the example 1 header

File: content/lua/concepts/strings/terms/format/format.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+```lua
+local name = "Angel"
+local height = 5.587
+local age = 20
+
+local formatted = string.format("Hello, my name is %s, I am %.1fr ft tall and I am %d years old.", name, height, age)
+print(formatted)
+```
+
+In this example, the string.format() function is used to create a formatted string by replacing `%s`, `%.1f` and `%d` placeholders with the values of name, age and height, respectively. The resulting formatted string is then printed to the console.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/format/format.md:49]:
```suggestion
```

File: content/lua/concepts/strings/terms/format/format.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+This will return the following output:
+
+```shell
+Hello, my name is Angel, I am 5.6 ft tall and I am 20 years old.
+```
+
+## Example 2
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/terms/format/format.md:57]:
```suggestion
## Example 2

The following example demonstrates the use of `string.format()` with a hexadecimal value.

```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@Brag2gr8 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by Brag2gr8** [comment]
@caupolicandiaz done with the changes

**Review by caupolicandiaz — APPROVED** [review]
@Brag2gr8 This is ready for a second review.

**Review by SSwiniarski — APPROVED** [review]
@Brag2gr8 Looks good.

**Pull request merged**